LeaveManagementModule/Backend/app/controllers/employee.js

```diff
@@ -7,10 +32,136 @@ var employee = {
     //Sample Method of Employee Controller
     'sample' : (req,res) => {
         res.send("Welcome to HRMS");
-    }
+    },
+    'sendEmail' : (req,res) => {
+        console.log(sGrid);
+        res.send(sGrid.send());
+    },
+    'createLeave' : (req,res) => {
+        if(!req.params.id)
+        { res.status(400).json({
+             message : "Employee Id Required",
+         }); 
+         } //Procedure for creating a request 
+         sql.connect(mssqlConfig).then(() => {
+             return sql.query("exec pConditionForLeave @EmployeeId= "+EmployeeId+", @LeaveId="+LeaveId+", @StartDate="+StartDate+", @EndDate="+EndDate+"; ")
+             
+         }).then(result => { 
+             res.status(201).json({"message" : "leave request created"});
+             console.dir(result)
+         }).catch(err => {
+             // ... error checks
+         })
+        sql.close();   //closing a database connection 
+    },
+    'getLeaves' : (req,res) => {
+      
+            if(!req.params.id)
+           { res.status(400).json({
+                message : "Employee Id Required",
+            }); 
+            } 
+           // View showing employee leave details
         
-}
-    
+           sql.connect(mssqlConfig).then(() => {
+           return sql.query("SELECT fUserLeaveDetail (@EmployeeId= "+EmployeeId+", @LeaveId="+LeaveId+") as result")
```

Comment:
This function returns request for a single leave
but the route is for getting all leave requests for a particular employee